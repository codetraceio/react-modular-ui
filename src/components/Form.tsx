import * as React from 'react';

import { getBlockName } from '../services/componentService';

export interface IFormProps {
  action?: string;
  method?: string;
  target?: string;
  children?: JSX.Element;

  name?: string;
  onChange?: (name: string, value: string, event: React.ChangeEvent<HTMLFormElement>) => void;
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}

export default function Form(props: IFormProps) {
  function onChange(event: React.ChangeEvent<HTMLFormElement>) {
    if (typeof props.onChange === 'function') {
      props.onChange(event.target.name, event.target.value, event);
    }
  }
  
  return (
    <form
      className={getBlockName('form')}
      action={props.action}
      method={props.method}
      name={props.name}
      target={props.target}
      onChange={onChange}
      onSubmit={props.onSubmit}
    >
      <input type="submit" style={{display: 'none'}} />
      {props.children}
    </form>
  );
}
