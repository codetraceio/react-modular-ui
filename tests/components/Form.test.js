import * as React from "react";
import * as renderer from "react-test-renderer";

import Form from "../../src/components/Form";

describe("Form", () => {
  it("should render Form", () => {
    const tree = renderer.create(
      <Form>Something</Form>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
