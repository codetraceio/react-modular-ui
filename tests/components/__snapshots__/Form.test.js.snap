// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render Form 1`] = `
<form
  className="c-form"
  onChange={[Function]}
>
  <input
    style={
      Object {
        "display": "none",
      }
    }
    type="submit"
  />
  Something
</form>
`;
